skyportal/handlers/api/photometry.py

```diff
@@ -204,18 +221,310 @@ def post(self):
+            phot_table['flux'] = df['flux'].fillna(np.nan)
+            phot_table['fluxerr'] = df['fluxerr'].fillna(np.nan)
+
+        # convert to microjanskies, AB for DB storage as a vectorized operation
+        pdata = PhotometricData(phot_table)
+        standardized = pdata.normalized(zp=PHOT_ZP, zpsys='ab')
+
+        df['standardized_flux'] = standardized.flux
+        df['standardized_fluxerr'] = standardized.fluxerr
+
+        df["hash"] = df.apply(hash_photometry, axis=1)
+
+        instcache = {}
+        for iid in df['instrument_id'].unique():
+            instrument = Instrument.query.get(int(iid))
+            if not instrument:
+                return self.error(f'Invalid instrument ID: {iid}')
+            instcache[iid] = instrument
+
+        for oid in df['obj_id'].unique():
+            obj = Obj.query.get(oid)
+            if not obj:
+                return self.error(f'Invalid object ID: {oid}')
+
+        # pre-fetch the photometry PKs. these are not guaranteed to be
+        # gapless (e.g., 1, 2, 3, 4, 5, ...) but they are guaranteed
+        # to be unique in the table and thus can be used to "reserve"
+        # PK slots for uninserted rows
+        pkq = (
+            f"SELECT nextval('photometry_id_seq') FROM "
+            f"generate_series(1, {len(df)})"
+        )
+
+        proxy = DBSession().execute(pkq)
+
+        # cache this as list for response
+        ids = [i[0] for i in proxy]
+        df['id'] = ids
+        rows = df.where(pd.notnull(df), None).to_dict('records')
+
+        params = []
+        for packet in rows:
+            if packet["filter"] not in instcache[packet['instrument_id']].filters:
+                raise ValidationError(
+                    f"Instrument {instrument.name} has no filter "
+                    f"{packet['filter']}."
+                )
+
+            flux = packet.pop('standardized_flux')
+            fluxerr = packet.pop('standardized_fluxerr')
+
+            # reduce the DB size by ~2x
+            keys = ['limiting_mag', 'magsys', 'limiting_mag_nsigma']
+            original_user_data = {key: packet[key] for key in keys if key in packet}
+            if original_user_data == {}:
+                original_user_data = None
+
+            phot = dict(
+                id=packet['id'],
+                original_user_data=original_user_data,
+                upload_id=upload_id,
+                flux=flux,
+                fluxerr=fluxerr,
+                obj_id=packet['obj_id'],
+                altdata=packet['altdata'],
+                instrument_id=packet['instrument_id'],
+                ra_unc=packet['ra_unc'],
+                dec_unc=packet['dec_unc'],
+                mjd=packet['mjd'],
+                filter=packet['filter'],
+                ra=packet['ra'],
+                dec=packet['dec'],
+                origin=packet["origin"],
+                hash=packet["hash"],
+            )
+
+            params.append(phot)
+
```

Comment:
I think the "old" code with this logic got moved into the PUT handler below, from git's perspective, and this block marked as new instead of the other way around